runtime/parachains/src/paras/benchmarking.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+use crate::{configuration::HostConfiguration, shared};
+use frame_benchmarking::{benchmarks, impl_benchmark_test_suite};
+use frame_system::RawOrigin;
+use primitives::v1::{HeadData, Id as ParaId, ValidationCode, MAX_CODE_SIZE, MAX_HEAD_DATA_SIZE};
+use sp_runtime::traits::{One, Saturating};
+
+// 2 ^ 10, because binary search time complexity is O(log(2, n)) and n = 1024 gives us a
+// big and round number.
+const SAMPLE_SIZE: u32 = 1024;
```

Comment:
maybe we should say that due to the limited number of parachains, the number of pruning, upcoming upgrades and cooldowns shouldn't exceed this number ?